code/Control/UserDefinedFormController.php

```diff
@@ -393,17 +394,22 @@ public function process($data, $form)
                 $mergeFields = $this->getMergeFieldsMap($emailData['Fields']);
 
                 if ($attachments && (bool) $recipient->HideFormData === false) {
-                    foreach ($attachments as $file) {
+                    foreach ($attachments as $uploadFieldName => $file) {
                         /** @var File $file */
                         if ((int) $file->ID === 0) {
                             continue;
                         }
 
-                        $email->addAttachmentFromData(
-                            $file->getString(),
-                            $file->getFilename(),
-                            $file->getMimeType()
-                        );
+                        $canAttachFileForRecipient = true;
+                        $this->extend('updateCanAttachFileForRecipient', $canAttachFileForRecipient, $recipient, $uploadFieldName, $file);
```

Comment:
We might need to put the `UserDefinedForm` instance in here as well, if we can. I can definitely see a scenario where attachments for a field named "uploaded_files" is allowed for form A, but attachments for the same named field on form B aren't allowed to be sent.

Reply:
You can get the form from the recipient via `$recipient->Form()`. Does that suffice?

Reply:
Yup that works for me. 👍